doc/readme.md

```diff
@@ -0,0 +1,67 @@
+Document written by Rune Stubbe, December 14th 2018
+
+Pseudorandom notes on Unity.Mathematics
+==============================================
+Differences between Mathematics and HLSL
+-----------------------------------------
+Unity.Mathematics tries to follow familiar HLSL conventions whenever possible. Formulas and computations are generally more succintly expressed in a HLSL-like syntax with swizzles and free function than the traditional C# Unity Math library. Unity.Mathematics is not HLSL, but is in most instances trivially portable. The large common subset allows for code to directly shared between the two in many cases.
+Due to inherent limitations in C# and for consistency reasons there are some notable differences between Unity.Mathematics and HLSL.
+1. Type conversion rules:
+HLSL will implicitly perform narrowing conversions. This could for instance be a float2 assigned directly to an int2/bool2.
+C# requires explict casts to perform these kinds of potentially lossy conversions. We have chosen to follow the C# conventions here for consistency with how the conversions already function with C#'s builtin types int/float/etc.
+2. Floating point literals:
+In HLSL the default floating point literal precision is float, while it is double in C#. The `1.5` float literal in HLSL corresponds to `1.5f` in C# and the likewise the double literal `3.2L` in HLSL corresponds to `3.2` in C#.
+3. Half precision floating point: `half`/`halfN` are supported in Mathematics, but only as storage types that implicitly convert to `float`/`double` types and can be explicitly converted to from these types. No arithmetic operation are defined on them as opposed to the HLSL types. We chose to do this as these operations would involve implicit conversion to and from `half`. These operations would be expensive and we believe that expensive operations should not happen implicitly.
```

Comment:
PS4 has fast half/float conversion, and it's safe to say that any PC good enough for Unity has it, too. This is a fairly recent development as of 2020, but it's a fact.